src/frontend/src/app/pages/profile-edit/admin-personal-info-edit/admin-personal-info-edit.component.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+} from '@angular/core';
+import {
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+
+import { AdminProfileComponent } from '../admin-profile';
+import {
+  AdminProfile,
+  AdminProfileUpdate,
+  ProfileService,
+  UserRole,
+} from '~core/state';
+import {
+  FormFieldComponent,
+  InputDirective,
+  InputErrorComponent,
+  LabelComponent,
+  TooltipDirective,
+} from '~core/ui';
+import { ComponentChanges } from '~core/utils';
+
+export interface AdminProfileForm {
+  username: FormControl<string>;
+  bio: FormControl<string>;
+}
+
+@Component({
+  selector: 'app-admin-personal-info-edit',
+  standalone: true,
+  imports: [
+    ReactiveFormsModule,
+    FormFieldComponent,
+    LabelComponent,
+    InputErrorComponent,
+    InputDirective,
+    TooltipDirective,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <form [formGroup]="profileForm" (ngSubmit)="onSubmit()">
+      <app-form-field>
+        <app-label>Username</app-label>
+
+        <input appInput id="username" type="text" formControlName="username" />
+
+        <app-input-error key="required">
+          Username cannot be empty
+        </app-input-error>
+        <app-input-error key="minlength">
+          Username must have at least 3 characters
+        </app-input-error>
+      </app-form-field>
+
+      <app-form-field>
+        <app-label>Bio</app-label>
+
+        <textarea
+          appInput
+          id="bio"
+          type="text"
+          formControlName="bio"
+        ></textarea>
+
+        <app-input-error key="required">Bio cannot be empty</app-input-error>
+      </app-form-field>
+
+      <div class="flex items-center">
+        <a
+          title="Cancel your edits"
+          class="ml-auto mr-4"
+          (click)="cancelEdits()"
+        >
+          Cancel
+        </a>
+        <button
+          type="submit"
+          [appTooltip]="
+            profileForm.invalid ? 'Fix the validation errors' : null
+          "
+          [disabled]="profileForm.invalid"
+          class="btn"
+        >
+          Save
+        </button>
+      </div>
+    </form>
+  `,
+})
+export class AdminPersonalInfoEditComponent implements OnChanges {
+  @Input({ required: true })
+  public userProfile!: AdminProfile;
+
+  @Output()
+  public formSave = new EventEmitter<void>();
```

Comment:
Rename to `close`.

Reply:
set to "formClose" due to ESLint check: "Output bindings, including aliases, should not be named as standard DOM events"